### PR TITLE
ComiciViewer sites

### LINKED ISSUE #5863: [Site Request] bigcomics.jp
**Name of the website**
BigComics

**Website urls (examples below)**
- Site: https://bigcomics.jp/
- Manga List: https://bigcomics.jp/series/list
- Manga example: https://bigcomics.jp/series/4a917e09503ce
- Chapter online viewer example: https://bigcomics.jp/episodes/5664113d6d6ea

**Languages**
Japanese

**Additional details**
The site has DRM protection.

### PR DESCRIPTION
**Pull request by MikeZeDev**
Turns out some japanese connectors use the same viewer: ComiciViewer.

This PR is mostly slight modifications in YoungChampion , nothing fancy.

* MODIFY :  YoungChampion is now "template like"
* ADD : BigComics (Fixes https://github.com/manga-download/hakuneko/issues/5863)
* MODIFY : Yanmaga to extend YoungChampion

If more websites use them, gonna do a proper connector. I mainly did that because it will be easier when checking while coding Haruneko.

### CODE REVIEW

**Comment by Ever4engel** [comment]
why i can't see the code [here](https://github.com/manga-download/hakuneko/pull/5867/commits/cae6492097cae17405d54fbad6445209f733cd6d)?

**Comment by Ever4engel** [comment]
Why don't you just code a template? will be easier for you in the future.

**Comment by MikeZeDev** [comment]
No particular reasons 🤷 . I feel like validating a new template PR is more repulsive than a modified connector but maybe its only in my head ;)

The future is Haruneko so it wont be a problem, i make sure to check for similarities when porting connectors, as the philosophy is now decorators :)

File: src/web/mjs/connectors/YoungChampion.mjs [hunk 1]
```diff
@@ -18,29 +19,54 @@ export default class YoungChampion extends Connector {
...
-        const data = await this.fetchDOM(request, 'div.series-ep-list a');
-        return data.map(chapter => {
+        const data = await this.fetchDOM(request, this.queryChapter );
+        return data.map(element => {
+            const titleElement = element.querySelector(this.queryChapterTitle);
             return {
-                id : new URL(chapter.dataset['href']).pathname,
-                title : chapter.querySelector('span.series-ep-list-item-h-text').textContent.trim()
+                id : new URL(element.dataset['href']).pathname,
+                title: titleElement ? (titleElement.textContent ||titleElement.text).trim() : element.text.trim()
```
Comment (Sheepux) [inline src/web/mjs/connectors/YoungChampion.mjs:69]: inconsistent formating, add space after `||`

**Review by Sheepux — APPROVED** [review]
LGTM, minor change suggested

**Comment by 27136773** [comment]
[Missing in HaruNeko](https://github.com/search?q=repo%3Amanga-download%2Fharuneko%20%27bigcomics%27&type=code)

**Pull request merged**